From 5e4e11855887cbb330a941f0250c0da0538e120d Mon Sep 17 00:00:00 2001
From: Timo Reichl <thetredev@gmail.com>
Date: Wed, 23 Apr 2025 22:41:37 +0000
Subject: [PATCH] support/buildbot/Versioning: ignore git HEAD

The repo is included as submodule. This is just a workaround to get it
to configure at all.

Signed-off-by: Timo Reichl <thetredev@gmail.com>
---
 support/buildbot/Versioning | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/support/buildbot/Versioning b/support/buildbot/Versioning
index 3c2101c..cc26e36 100644
--- a/support/buildbot/Versioning
+++ b/support/buildbot/Versioning
@@ -16,21 +16,21 @@ outputs = [
   os.path.join(builder.buildFolder, 'includes', 'metamod_version_auto.h')
 ]
 
-with open(os.path.join(builder.sourcePath, '.git', 'HEAD'), encoding='utf-8') as fp:
-  head_contents = fp.read().strip()
-  if re.search('^[a-fA-F0-9]{40}$', head_contents):
-    git_head_path = os.path.join(builder.sourcePath, '.git', 'HEAD')
-  else:
-    git_state = head_contents.split(':')[1].strip()
-    git_head_path = os.path.join(builder.sourcePath, '.git', git_state)
-    if not os.path.exists(git_head_path):
-      git_head_path = os.path.join(builder.sourcePath, '.git', 'HEAD')
+# with open(os.path.join(builder.sourcePath, '.git', 'HEAD'), encoding='utf-8') as fp:
+#   head_contents = fp.read().strip()
+#   if re.search('^[a-fA-F0-9]{40}$', head_contents):
+#     git_head_path = os.path.join(builder.sourcePath, '.git', 'HEAD')
+#   else:
+#     git_state = head_contents.split(':')[1].strip()
+#     git_head_path = os.path.join(builder.sourcePath, '.git', git_state)
+#     if not os.path.exists(git_head_path):
+#       git_head_path = os.path.join(builder.sourcePath, '.git', 'HEAD')
 
 sources = [
   os.path.join(builder.sourcePath, 'product.version'),
 
   # This is a hack, but we need some way to only run this script when Git changes.
-  git_head_path,
+  #git_head_path,
 
   # The script source is a dependency, of course...
   argv[1]
-- 
2.49.0

